state/core_access.go

```diff
@@ -174,24 +194,70 @@ func (ca *CoreAccessor) SubmitPayForBlob(
 		appblobs[i] = &b.Blob
 	}
 
-	response, err := appblob.SubmitPayForBlob(
-		ctx,
-		ca.signer,
-		ca.coreConn,
-		appblobs,
-		apptypes.SetGasLimit(gasLim),
-		withFee(fee),
-	)
-	// metrics should only be counted on a successful PFD tx
-	if err == nil && response.Code == 0 {
-		ca.lastPayForBlob = time.Now().UnixMilli()
-		ca.payForBlobCount++
+	// if the min gas price is not set we query the node that we are connected for it
+	minGasPrice, err := ca.getMinGasPrice(ctx)
+	if err != nil {
+		return nil, err
 	}
 
-	if response != nil && response.Code != 0 {
-		err = errors.Join(err, sdkErrors.ABCIError(response.Codespace, response.Code, response.Logs.String()))
+	// we only estimate gas if the user wants us to (by setting the gasLim to 0). In the future we may want
+	// to make these arguments optional.
+	if gasLim == 0 {
+		blobSizes := make([]uint32, len(blobs))
+		for i, blob := range blobs {
+			blobSizes[i] = uint32(len(blob.Data))
+		}
+
+		// TODO: the default gas per byte and the default tx size cost per byte could be changed through governance
```

Comment:
Pls add you handle to the TODO

Reply:
Can I do it in a separate PR

Reply:
I am asking to make it `//TODO(@cmwaters)`. Why a separate PR?

Reply:
ah yup sorry will do